### PR TITLE
RUMM-1931 stopTrackingDatadogEvents added to avoid memory leaks

### PR DESCRIPTION
**Pull request by buranmert**
### What and why?

Without removing `WKScriptMessageHandlers` explicitly, they stay alive in memory even after deallocation of the webview

### How?

By calling `stopTrackingDatadogEvents`, our users can remove Datadog components from their `WKWebViewConfiguration` instances so that `DatadogMessageHandler` can be deallocated.

### Review checklist

- [x] Feature or bugfix MUST have appropriate tests (unit, integration)
- [x] Make sure each commit and the PR mention the Issue number or JIRA reference

### CODE REVIEW

File: Sources/Datadog/FeaturesIntegration/WebView/WKUserContentController+Datadog.swift [hunk 1]
```diff
@@ -27,6 +27,22 @@ public extension WKUserContentController {
         addDatadogMessageHandler(allowedWebViewHosts: hosts, hostsSanitizer: HostsSanitizer())
     }
 
+    /// Disables Datadog iOS SDK and Datadog Browser SDK integration.
+    ///
+    /// Removes Datadog's ScriptMessageHandler and UserScript from the caller.
+    /// _NOTE:_ This method **must** be called when the webview can be deinitialized.
```
Comment (maxep) [inline Sources/Datadog/FeaturesIntegration/WebView/WKUserContentController+Datadog.swift:33]: This won't render properly, better to use
```suggestion
    /// - Note: This method **must** be called when the webview can be deinitialized.
```

File: Tests/DatadogTests/Datadog/RUM/WebView/WKUserContentController+DatadogTests.swift [hunk 2]
```diff
@@ -23,6 +23,12 @@ final class DDUserContentController: WKUserContentController {
     }
 }
 
+final class MockMessageHandler: NSObject, WKScriptMessageHandler {
+    func userContentController(_ userContentController: WKUserContentController, didReceive message: WKScriptMessage) {
+        return
+    }
```
Comment (maxep) [inline Tests/DatadogTests/Datadog/RUM/WebView/WKUserContentController+DatadogTests.swift:29]: /nit
```suggestion
    func userContentController(_ userContentController: WKUserContentController, didReceive message: WKScriptMessage) { }
```

File: Tests/DatadogTests/Datadog/RUM/WebView/WKUserContentController+DatadogTests.swift [hunk 3]
```diff
@@ -96,6 +102,31 @@ class WKUserContentController_DatadogTests: XCTestCase {
         XCTAssertEqual(recordedLogMessages, Array(repeating: "`trackDatadogEvents(in:)` was called more than once for the same WebView. Second call will be ignored. Make sure you call it only once.", count: multipleTimes - 1))
     }
 
+    func testWhenStoppingTracking_itKeepsNonDatadogComponents() throws {
+        let controller = DDUserContentController()
+
+        controller.trackDatadogEvents(in: [])
+
+        let componentCount = 10
```
Comment (maxep) [inline Tests/DatadogTests/Datadog/RUM/WebView/WKUserContentController+DatadogTests.swift:110]: why make it 10? I guess adding a single or a couple of other scripts will cover this change

**Review by maxep — APPROVED** [review]
LGTM! I've minor suggestions tho

**Review by ncreated — APPROVED** [review]
👍 It would be also good to add `stopTrackingDatadogEvents()` in "Webviews integration" screen in Example app. Otherwise we don't have any use of this API outside unit tests.

**Pull request merged**